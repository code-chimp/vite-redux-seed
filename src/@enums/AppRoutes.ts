enum AppRoutes {
  Home = '/',
  Users = '/users',
  Notifications = '/notifications',
  Fallthrough = '*',
}

export default AppRoutes;
