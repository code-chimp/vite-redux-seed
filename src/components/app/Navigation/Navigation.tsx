/* istanbul ignore file */
/* no logic */
import React, { FC } from 'react';
import { Link } from 'react-router-dom';
import AppRoutes from '../../../@enums/AppRoutes';

const Navigation: FC = () => {
  return (
    <ul className="nav">
      <li className="nav-item">
        <Link className="nav-link" to={AppRoutes.Home}>
          Home
        </Link>
      </li>
      <li className="nav-item">
        <Link className="nav-link" to={AppRoutes.Users}>
          Users
        </Link>
      </li>
      <li className="nav-item">
        <Link className="nav-link" to={AppRoutes.Notifications}>
          Notifications
        </Link>
      </li>
    </ul>
  );
};

export default Navigation;
