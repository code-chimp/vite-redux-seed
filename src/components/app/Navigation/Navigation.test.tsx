import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import AppRoutes from '../../../@enums/AppRoutes';
import Navigation from './Navigation';

describe('components / app / Navigation', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={[AppRoutes.Users]}>
        <Navigation />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
