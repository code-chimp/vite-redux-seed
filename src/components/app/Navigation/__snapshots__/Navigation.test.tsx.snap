// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components / app / Navigation > should match the snapshot 1`] = `
<DocumentFragment>
  <ul
    class="nav"
  >
    <li
      class="nav-item"
    >
      <a
        class="nav-link"
        href="/"
      >
        Home
      </a>
    </li>
    <li
      class="nav-item"
    >
      <a
        class="nav-link"
        href="/users"
      >
        Users
      </a>
    </li>
    <li
      class="nav-item"
    >
      <a
        class="nav-link"
        href="/notifications"
      >
        Notifications
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
